### PR TITLE
ref(aci): pass WorkflowJob into process_workflows

### PR DESCRIPTION
**Pull request by cathteng**
Original PR https://github.com/getsentry/sentry/pull/82096/

`process_workflows` is currently called with a `GroupEvent` for issue types like metric issues to fire downstream actions.

To process issue alerts, we use other elements of `PostProcessJob` in `post_process` to evaluate conditions. As such, we should pass a version of `PostProcessJob` (here called `WorkflowJob`) into `process_workflows`. It contains a required `GroupEvent` and then makes everything else from `PostProcessJob` optional.

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -1002,10 +1003,19 @@ def process_workflow_engine(job: PostProcessJob) -> None:
...
+    # PostProcessJob event is optional, WorkflowJob event is required
+    if "event" not in job:
+        logger.error("Missing event to create WorkflowJob", extra={"job": job})
+        return
+
+    try:
+        workflow_job = WorkflowJob({**job})  # type: ignore[typeddict-item]
+    except Exception:
+        logger.exception("Could not create WorkflowJob", extra={"job": job})
+        return
```
Comment (cathteng) [inline src/sentry/tasks/post_process.py:1015]: couldn't make mypy happy with this, it kept complaining the `event` didn't exist

File: src/sentry/workflow_engine/types.py [hunk 2]
```diff
@@ -28,9 +29,21 @@ class DetectorPriorityLevel(IntEnum):
...
+
+
+class WorkflowJob(EventJob, total=False):
+    group_state: GroupState
+    is_reprocessed: bool
+    has_reappeared: bool
+    has_alert: bool
+    has_escalated: bool
+
+
```
Comment (cathteng) [inline src/sentry/workflow_engine/types.py:43]: the main piece of this PR

File: src/sentry/workflow_engine/handlers/condition/group_state_handlers.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+@condition_handler_registry.register(Condition.REGRESSED_EVENT)
+class RegressedEventConditionHandler(DataConditionHandler[WorkflowJob]):
+    @staticmethod
+    def evaluate_value(job: WorkflowJob, comparison: Any) -> bool:
+        state = job.get("group_state", None)
+        if state is None:
+            return False
+
+        return state["is_regression"] == comparison
```
Comment (ceorourke) [inline src/sentry/workflow_engine/handlers/condition/group_state_handlers.py:16]: is `is_regression` guaranteed to exist on `state`? if not I'd do a `.get()`

Comment (cathteng) [reply src/sentry/workflow_engine/handlers/condition/group_state_handlers.py:16]: yes it's guaranteed, the state is a typed dict too https://github.com/getsentry/sentry/blob/a62dc3bceb196b2caa327e2941bad15980d717cc/src/sentry/eventstream/base.py#L30-L34

**Review by ceorourke — APPROVED** [review]

**Pull request merged**